npm/pkg/dataplane/dpshim/dpshim.go

```diff
@@ -143,8 +144,13 @@ func (dp *DPShim) deleteIPSet(setMetadata *ipsets.IPSetMetadata) {
 }
 
 func (dp *DPShim) AddToSets(setMetadatas []*ipsets.IPSetMetadata, podMetadata *dataplane.PodMetadata) error {
-	dp.Lock()
-	defer dp.Unlock()
+	if len(setMetadatas) == 0 {
+		return nil
+	}
+
+	dp.lock()
+	defer dp.unlock()
+
```

Comment:
needs to be updated to be like ipsetmanager.go
```
		_, ok := set.IPPodKey[ip]
		set.IPPodKey[ip] = podKey
		if ok {
			continue
		}
```

Reply:
or perhaps this, because it needs to share when the podkey has changed
```
		oldPodKey, ok := set.IPPodKey[ip]
		if ok && oldPodKey == podKey {
			continue
		}
		set.IPPodKey[ip] = podKey
```